rust/worker/src/execution/operators/brute_force_knn.rs

```diff
@@ -101,27 +161,28 @@ impl Operator<BruteForceKnnOperatorInput, BruteForceKnnOperatorOutput> for Brute
         };
 
         let mut heap = BinaryHeap::with_capacity(input.k);
-        let data_chunk = &input.data;
+        let data_chunk = logs;
         for data in data_chunk.iter() {
             let log_record = data.0;
             let index = data.1;
 
-            if log_record.record.operation == Operation::Delete {
+            if log_record.final_operation == Operation::Delete {
                 // Explicitly skip deleted records.
                 continue;
             }
 
             // Skip records that are disallowed. If allowed list is empty then
             // don't exclude anything.
             // Empty allowed list is passed when where filtering is absent.
+            // TODO: This should not need to use merged_user_id, which clones the id.
             if !input.allowed_ids.is_empty()
                 && !input
                     .allowed_ids_brute_force
-                    .contains(&log_record.record.id)
+                    .contains(&log_record.merged_user_id())
             {
                 continue;
             }
-            let embedding = match &log_record.record.embedding {
+            let embedding = match &log_record.final_embedding {
```

Comment:
Shouldn't this be `log_record.merged_embeddings()`? Embeddings can be none in the Update and only present in the segment

Reply:
OK. Got confused by the interface.